plugins/pulp_python/plugins/importers/sync.py

```diff
@@ -39,117 +39,103 @@ def __init__(self, *args, **kwargs):
-                    url = package['url']
-                    self.parent.available_units.append(unit)
-                    self.parent.unit_urls[unit] = url
+                unit = models.Package.from_json(package, version, metadata['info'])
+                self.parent.available_units.append(unit)
 
 
 class DownloadPackagesStep(publish_step.DownloadStep):
     """
-    This DownloadStep retrieves the packages from the feed, processes each package for its metadata,
-    and adds the unit to the repository in Pulp.
+    Retrieves package bits for each request. Once the bits are successfully downloaded, the
+    cooresponding unit (models.Package) is saved into the database and associated to the repository.
     """
 
     def download_succeeded(self, report):
         """
-        This method processes a downloaded Python package. It opens the package and reads its
-        PKG-INFO metadata file to determine all of its metadata. This step can be slow for larger
-        packages since it needs to decompress them to do this. Despite it being slower to do this
-        than it would be to read the metadata from the metadata file we downloaded earlier, we do
-        not get the metadata for older versions from that file. Thus, this is the only reliable way
-        to represent the metadata for different versions of a package. It also has the benefit of
-        code reuse for determining the metadata, as the upload code also acquires the metadata this
-        way.
-
-        This method also ensures that the checksum of the downloaded package matches the checksum
-        that was listed in the manifest. If everything checks out, the package is added to the
-        repository and moved to the proper storage path.
+        When the package's bits are successfully downloaded, the checksum is verified, the package
+        is moved to its final location, and the models.Package object is saved into the database
+        and associated to the repository.
 
         :param report: The report that details the download
         :type  report: nectar.report.DownloadReport
         """
         _logger.info(_('Processing package retrieved from %(url)s.') % {'url': report.url})
-
-        checksum = models.Package.checksum(report.destination, report.data._checksum_type)
-        if checksum != report.data._checksum:
+        package = report.data
+        checksum = models.Package.checksum(report.destination, package._checksum_type)
+        if checksum != package._checksum:
             report.state = 'failed'
-            report.error_report = {'expected_checksum': report.data._checksum,
+            report.error_report = {'expected_checksum': package._checksum,
                                    'actual_checksum': checksum}
             return self.download_failed(report)
 
-        package = models.Package.from_archive(report.destination)
+        # Unless checksum from upstream is default type, recalculate checksum
```

Comment:
I'm a little surprised by this behavior. Doesn't this mean ``DEFAULT_CHECKSUM_TYPE`` is more like ``CHECKSUM_TYPE``? I expect a default to be used if nothing else is available, but not if there is something else set.